LanguageFeatures/Patterns/list_A05_t06.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// listPattern         ::= typeArguments? '[' listPatternElements? ']'
+/// listPatternElements ::= listPatternElement ( ',' listPatternElement )* ','?
+/// listPatternElement  ::= pattern | restPattern
+/// restPattern         ::= '...' pattern?
+///
+/// A list pattern may contain a rest element which allows matching lists of
+/// arbitrary lengths. If a rest element is present and has a subpattern, all of
+/// the elements not matched by other subpatterns are collected into a new list
+/// and that list is matched against the rest subpattern.
+///
+/// var [a, b, ...rest, c, d] = [1, 2, 3, 4, 5, 6, 7];
+/// print('$a $b $rest $c $d'); // Prints "1 2 [3, 4, 5] 6 7".
+///
+/// @description Check that constant subpattern can be used in a list's rest
+/// pattern
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "patterns_lib.dart" as p;
+import "../../Utils/expect.dart";
+
+class C {
+  static const c = 42;
+  const C();
+}
+
+String test1(List list) {
+  return switch (list) {
+    case <int>[1, 2, 3, 4, ... 42] => "case-1";
```

Comment:
OK, let's do that.

I'm working on an issue that raises several issues, including the fact that `... 42` will _definitely_ never succeed, but for now we'll just assume that it is accepted (and the match will fail at run time, but then we just continue to the next case, so it isn't going to be obvious that it's a "bad" pattern).